gnomad_qc/v4/create_release/variant_statistics.py

```diff
@@ -0,0 +1,286 @@
+#     default_generate_gene_lof_summary,
+#     get_summary_counts,
+# )
+from gnomad.resources.resource_utils import (
+    TableResource,
+    VersionedTableResource,
+)
+
+# from gnomad.utils.filtering import filter_to_adj
+from gnomad.utils.slack import slack_notifications
+
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v4.resources.basics import (
+    calling_intervals,
+    get_gnomad_v4_vds,
+)
+import gnomad_qc.v4.resources.meta as meta
+from gnomad_qc.v4.resources.release import (
+    release_sites,
+    CURRENT_RELEASE,
+    RELEASES,
+    _release_root,
+)
+
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("release")
+logger.setLevel(logging.INFO)
+
+
+def get_per_sample_variant_counts(
+    test: bool = False, data_type: str = "exomes"
+) -> VersionedTableResource:
+    """
+    Get per-sample variant counts and other information as generated by Hail's Sample QC module.
+
+    :param test: Whether to use a tmp path for analysis of the test VDS instead of the full v4 VDS
+    :param data_type: Data type used in sample QC, e.g. "exomes" or "genomes"
+    :return: Per sample variant counts table
+    """
+    return VersionedTableResource(
+        CURRENT_RELEASE,
+        {
+            version: TableResource(
+                f"{_release_root(version=version, test=test, data_type=data_type, extension='ht')}/gnomad.{data_type}.v{version}.per_sample_variant_counts.ht"
+            )
+            for version in RELEASES
+        },
+    )
+
+
+def per_sample_datatype(
+    data_type: str = "exomes",
+    test: bool = False,
+    overwrite: bool = False,
+    filter_variant_qc: bool = False,
+    filter_calling_intervals: bool = False,
+    singletons_by_pop: bool = False,
+) -> None:
+    """
+    Write out Hail's sample_qc output for desired samples and variants.
+    Useful for finding the amount of variants (total or specific types) per sample.
+    Also prints distribution of total variants called per sample.
+
+    :param data_type: String of either "exomes" or "genomes" for the sample type.
+    :param test: Boolean for if you would like to use a small test set.
+    :param overwrite: Boolean to overwrite checkpoint files if requested.
+    :param filter_variant_qc: Bool to only count variants that pass all variant qc filters (i.e. VQSR, AC0).
+    :param filter_calling_intervals: Bool to filter only to variants in calling intervals.
+    :param singletons_by_pop: Bool to calculate the mean number of singletons by population group.
+    """
+
+    # Read in release data and metadata and VDS
+    ht_release_data = release_sites(data_type).ht()
+    if test:
+        logger.info("Test: filtering to only chr22 sites and 467 test samples in VDS")
+        ht_release_data = ht_release_data.filter(
+            ht_release_data.locus.contig == "chr22"
+        )
+
+    if filter_variant_qc:
```

Comment:
Instead of having this as an option, lets just add it as a stratification (so I want a number with no filtering and a number with filtering

Reply:
I believe this would require running hl.vds.sample_qc() twice , once on a VDS filtered to this and once without. Is that something we'd want to support within the script, or just have users run twice ? We can put a pin in this for our chat Wednesday if you'd like